### ISSUE TITLE
Can't update issue, get "Specify the Priority (name) in the string format"

### ISSUE THREAD

**Issue by tillydray** [ahungry/org-jira#313]
Thank you for allowing me to do Jira in Emacs, it's so nice!

I created an issue via org-jira and it worked fine. I can see the issue in the browser.

I want to update it's description so I navigate to NSR.org where all my issues are. I add more info to the description (see `ticket` below), save the buffer, and do `org-jira-update-issue` but I get an error that says `error in process sentinel: JIRA_ERROR - see your *Messages* buffer for more details.` See `output from Messages` below.

<details>
  <summary>ticket</summary>
  
```org-mode
** TODO REDACTED :NSR_864:
:PROPERTIES:
:assignee:
:filename: NSR-864
:reporter: Me
:type:     Chore
:type-id:  10040
:priority: High
:status:   Open
:created:  2023-01-10T11:37:45.815-0500
:updated:  2023-01-10T11:37:45.815-0500
:ID:       NSR-864
:CUSTOM_ID: NSR-864
:END:
:LOGBOOK:
:END:
*** description: [[REDACTED][NSR-864]]
  REDACTED_ORIGINAL_TEXT

  REDACTED_NEW_TEXT
```

</details>

<details>
  <summary>output from Messages</summary>

```
"JIRA_ERROR REQUEST: "

"/rest/api/2/issue/NSR-864"

(:type "PUT" :data "{\"fields\":{\"components\":[],\"labels\":[\"NSR-864\"],\"priority\":{\"id\":null,\"name\":\"NSR-864\"},\"description\":\"REDACTED",\"assignee\":{\"id\":null},\"reporter\":{\"id\":null},\"summary\":\"REDACTED\",\"issuetype\":{\"id\":\"NSR-864\",\"name\":\"NSR-864\"}}}")

"JIRA_ERROR RESPONSE: "

((errorMessages . []) (errors (priority . "Specify the Priority (name) in the string format")))

error in process sentinel: JIRA_ERROR - see your *Messages* buffer for more details. [2 times]
```
</details>

In case it's relevant, the REDACTED_NEW_TEXT is visible in the error output so it's getting sent

**Comment by ahungry** [comment]
Try to `M-x customize` `jiralib-update-issue-fields-exclude-list` and ensure priority is set as excluded.

Full source of the custom:
```
(defcustom jiralib-update-issue-fields-exclude-list nil
  "A list of symbols to check for exclusion on updates based on matching key.
Key names should be one of components, description, assignee, reporter, summary, issuetype."
  :type '(set (const :tag "Exclude components" components)
              (const :tag "Exclude description" description)
              (const :tag "Exclude assignee" assignee)
              (const :tag "Exclude reporter" reporter)
              (const :tag "Exclude summary" summary)
              (const :tag "Exclude priority" priority)
              (const :tag "Exclude issue type" issuetype))
  :group 'org-jira)
```

It should avoid attempting to sync a field that your Jira may not have enabled on the issue (priority).

**Comment by tillydray** [comment]
@ahungry thanks for the quick reply! That fixed that problem, and I also had to add `issuetype` and `reporter` to the list but it's working now. Thanks!

I'm curious what I could have done to figure this out on my own. I did all I knew to do, but the error message wasn't helpful enough for me. I thought it was telling me to add a "name" field to "priority" but it was already there `\"priority\":{\"id\":null,\"name\":\"NSR-864\"}`. Any tips for future me?
